### PR TITLE
[2nd PR] Addition of velocity_control actions and support for Gaussian action

### PR DESCRIPTION
**Pull request by naokiyokoyama**
## Motivation and Context

This change allows the agent to move forward and turn simultaneously using habitat-sim's VelocityControl physics object for calculating end positions given a time step, linear velocity, and angular velocity. Support has also been added such that the policy can now output continuous actions, using a Gaussian action distribution. 

## How Has This Been Tested

Train and evaluation can be done the same as before; the line

    action_distribution_type: "gaussian"

just needs to be added under RL.POLICY in the config yaml. The changes made to ppo_trainer.py will then automatically adjust the policy to output a Gaussian distribution, and if training, will make changes to the RolloutStorage to save float value previous actions (rather than long for discrete) and to save two values (lin vel, ang vel) rather than one (discrete action index). I trained for about 100M frames and evaluated the checkpoints successfully for both gaussian and categorical (original) distributions, with the only difference between the runs being the addition or deletion of the aforementioned line to the config yaml. 

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

There was an issue where sometimes the quaternions from get_agent_state were quantized too much such that its norm was off by about 1e-4 from a value of 1.0, which caused errors since habitat-sim only allows deviations of 1e-5.

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]
Looks good to m. Thoughts @erikwijmans ? @mathfac

File: habitat_baselines/utils/common.py [hunk 1]
```diff
@@ -518,3 +571,21 @@ def create_tar_archive(archive_path: str, dataset_path: str) -> None:
 
 def delete_folder(path: str) -> None:
     shutil.rmtree(path)
+
+
+def action_to_velocity_control(
```
[the remark this answers is not in the document]
Comment (naokiyokoyama) [reply habitat_baselines/utils/common.py:576]: I had tried to do that before, but ran into this error:

` habitat_baselines/rl/ppo/ppo_trainer.py:1009: error: Argument 1 to "step" of "VectorEnv" has incompatible type "List[Dict[str, Any]]"; expected "List[Union[int, str, Dict[str, Any]]]"  [arg-type]
  habitat_baselines/rl/ppo/ppo_trainer.py:1009: note: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
  habitat_baselines/rl/ppo/ppo_trainer.py:1009: note: Consider using "Sequence" instead, which is covariant`

Do you know what I could change the output type to, or if there's some way to ignore this error?
  
image

File: habitat_baselines/utils/common.py [hunk 2]
```diff
@@ -75,6 +76,58 @@ def forward(self, x: Tensor) -> CustomFixedCategorical:
         return CustomFixedCategorical(logits=x)
 
 
+class CustomNormal(torch.distributions.normal.Normal):
+    def sample(
+        self, sample_shape: Size = torch.Size()  # noqa: B008
+    ) -> Tensor:
+        return super().rsample(sample_shape)
+
+    def log_probs(self, actions):
```
[the remark this answers is not in the document]
Comment (naokiyokoyama) [reply habitat_baselines/utils/common.py:85]: Fixed, thank you

File: habitat/tasks/nav/nav.py [hunk 3]
```diff
@@ -40,8 +40,16 @@
 
 try:
     from habitat.sims.habitat_simulator.habitat_simulator import HabitatSim
+    from habitat_sim.bindings import RigidState
```
[the remark this answers is not in the document]
Comment (naokiyokoyama) [reply habitat/tasks/nav/nav.py:44]: Fixed, thank you

File: habitat_baselines/utils/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Skylion007) [reply habitat_baselines/utils/common.py:576]: Yeah, List has to be invariant, Can you type it as a Sequence instead?

[the remark this answers is not in the document]
Comment (naokiyokoyama) [reply habitat_baselines/utils/common.py:576]: Done, made it a Sequence instead.

**Review by erikwijmans — APPROVED** [review]
This looks good to me.  There is some new functionality needed in habitat-baselines to cleanly support other action spaces, but that is beyond the scope of this

**Pull request merged**